### PR TITLE
mrc-1713 wire up front-end ADR key widget

### PR DESCRIPTION
**Pull request by hillalex**
Contains commits from https://github.com/mrc-ide/hint/pull/276 which should be merged first and the base of this PR put back to master.

This PR wires up the ADR API key widget so it actually fetches/saves/removes the key from the backend.

### CODE REVIEW

File: src/app/static/src/app/store/root/actions.ts [hunk 1]
```diff
@@ -52,5 +57,27 @@ export const actions: ActionTree<RootState, RootState> & RootActions = {
 
     async changeLanguage(context, payload) {
         await changeLanguage<RootState>(context, payload)
+    },
+
+    async fetchADRKey(context) {
+        await api<RootMutation, RootMutation>(context)
+            .withSuccess(RootMutation.UpdateADRKey)
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/root/actions.ts:64]: Should include ignoreError to suppress console warning? Or do something on error?

File: src/app/static/src/app/store/root/actions.ts [hunk 2]
```diff
@@ -52,5 +57,27 @@ export const actions: ActionTree<RootState, RootState> & RootActions = {
...
+    async saveADRKey(context, key) {
+        await api<RootMutation, RootMutation>(context)
+            .withSuccess(RootMutation.UpdateADRKey)
+            .postAndReturn("/adr/key/", qs.stringify({key}));
+    },
+
+    async deleteADRKey(context) {
+        await api<RootMutation, RootMutation>(context)
+            .delete("/adr/key/")
+            .then((response) => {
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/root/actions.ts:77]: Why not withSuccess as for the others?

File: src/app/static/src/app/store/root/actions.ts (hunk 1 above)
Comment (hillalex) [reply src/app/static/src/app/store/root/actions.ts:64]: Probably should handle this error actually and show back to the user :+1:

File: src/app/static/src/app/store/root/actions.ts (hunk 2 above)
Comment (hillalex) [reply src/app/static/src/app/store/root/actions.ts:77]: I was thinking the payload back from the API would be "OK", wile the payload we want to commit is `null`. But actually looking at the controller the payload back from the API will be null, so this _should_ work as `withSuccess`, will change that over.

**Comment by hillalex** [comment]
Codecov at the project level complaining, no idea why tbh. I'll turn that off as we've done on other projects, as it's really the diff coverage we should be considering.

File: src/app/static/src/app/store/root/actions.ts [hunk 3]
```diff
@@ -52,5 +57,28 @@ export const actions: ActionTree<RootState, RootState> & RootActions = {
 
     async changeLanguage(context, payload) {
         await changeLanguage<RootState>(context, payload)
+    },
+
+    async fetchADRKey(context) {
+        await api<RootMutation, RootMutation>(context)
+            .ignoreErrors()
```
Comment (EmmaLRussell) [inline src/app/static/src/app/store/root/actions.ts:64]: Does this still need to be changed to withError?

Comment (hillalex) [reply src/app/static/src/app/store/root/actions.ts:64]: I think if fetching the API key fails there's not much point in complaining about it as the user can't really take another action to rectify it? Unlike saving/deleting where they can try again.

Comment (EmmaLRussell) [reply src/app/static/src/app/store/root/actions.ts:64]: Fair enough!

**Pull request merged**